src/main/scala/com/advancedtelematic/director/manifest/ManifestCompiler.scala

```diff
@@ -0,0 +1,89 @@
+package com.advancedtelematic.director.manifest
+
+import cats.data.ValidatedNel
+import com.advancedtelematic.director.data.UptaneDataType.Image
+import com.advancedtelematic.director.data.DbDataType.{Assignment, DeviceKnownStatus, EcuTarget, EcuTargetId}
+import com.advancedtelematic.director.data.DeviceRequest.{DeviceManifest, EcuManifest}
+import com.advancedtelematic.director.http.Errors
+import com.advancedtelematic.libats.data.DataType.{Checksum, HashMethod, Namespace}
+import com.advancedtelematic.libats.data.EcuIdentifier
+import org.slf4j.LoggerFactory
+
+import scala.util.{Failure, Success, Try}
+
+object ManifestCompiler {
```

Comment:
I am not sure about the name "Compiler", since it takes a Manifest as input and doesn't create it or something. `ManifestValidator` or `ValidatedManifest` maybe?